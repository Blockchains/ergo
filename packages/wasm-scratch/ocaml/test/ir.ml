open Ergo_wasm.Ir

let f_nop = func [nop]

let f_nop = func [call f_nop]

let g_alloc_p = global ~mutable_:true i32 [i32_const' 0]

let f_add_i32 = func ~params:[i32; i32] ~result:[i32]
    [ local_get 0
    ; local_get 1
    ; i32_add
    ]

let m =
  { start = Some f_nop
  ; funcs = ["add_i32", f_add_i32]
  ; globals = ["alloc_p", g_alloc_p]
  ; data = [0, "hello world"]
  }

let () =
  Wasm.Print.module_ stdout 72 (compile m)
