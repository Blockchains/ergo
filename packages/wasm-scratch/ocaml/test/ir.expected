(module
  (type $0 (func (param i32 i32) (result i32)))
  (type $1 (func))
  (memory $0 1)
  (global $0 (mut i32) (i32.const 0))
  (func $0 (type 0) (local.get 0) (local.get 1) (i32.add))
  (func $1 (type 1) (nop))
  (func $2 (type 1) (call 1))
  (export "memory" (memory 0))
  (export "alloc_p" (global 0))
  (export "add_i32" (func 0))
  (start 2)
  (data 0 (offset (i32.const 0)) "\68\65\6c\6c\6f\20\77\6f\72\6c\64")
)
